governator-core/src/main/java/com/netflix/governator/visitors/BindingTracingVisitor.java

```diff
@@ -0,0 +1,40 @@
+package com.netflix.governator.visitors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Binding;
+import com.google.inject.spi.DefaultElementVisitor;
+import com.netflix.governator.ModuleBuilder;
+
+/**
+ * Visitor for logging the entire binding information for each Element
+ * 
+ * To use with {@link ModuleBuilder},
+ * 
+ * <code>
+ * ModuleBuilder
+ *      .withModules(new MyApplicationModule)
+ *      .forEachElement(new BindingTracingVisitor())
+ *      .createInjector();
+ * </code>
+ */
+public class BindingTracingVisitor extends DefaultElementVisitor<Void> {
+    private static final Logger LOG = LoggerFactory.getLogger(ModuleBuilder.class);
+    
+    private final String prefix;
+    
+    public BindingTracingVisitor() {
+        this("");
```

Comment:
Should this be null? Otherwise the prefix will be (according to the logic in the second constructor) " : ".